src/classification_heads.py

```diff
@@ -1,31 +1,40 @@
-import tensorflow as tf
+import tensorflow.compat.v1 as tf
+tf.disable_v2_behavior()
+
 from flip_gradient import flip_gradient
 from gradient_projection import gradient_projection
 
 
 def regular(y, config):
-    if config['model_number'] == 20:
-        # VGGish models already defined their classificaton head
+    if config["model_number"] == 20:
+        # VGGish models already defined their classification head
         return y
     else:
-        return tf.compat.v1.layers.dense(inputs=y,
-                                    activation=None,
-                                    units=config['num_classes_dataset'],
-                                    kernel_initializer=tf.contrib.layers.variance_scaling_initializer(seed=config['seed']))
+        initializer = tf.keras.initializers.VarianceScaling(scale=2.0, seed=config["seed"])
```

Comment:
To keep the behaviour of the [VarianceScaling](https://www.tensorflow.org/api_docs/python/tf/keras/initializers/VarianceScaling) initializer we should set the parameter `distribution` to `untruncated_normal`, which was the default behaviour of `tf.contrib.layers.variance_scaling_initializer`. Although on informal testing `truncated_normal` seems to be helping on training and it is interesting to explore its impact.

Reply:
I might be missing something, but I understood the [documentation](https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/contrib/layers/variance_scaling_initializer) exactly in the opposite way:

> When initializing a deep network, it is in principle advantageous to keep the scale of the input variance constant, so it does not explode or diminish by reaching the final layer. This initializer use the following formula:
```  if mode='FAN_IN': # Count only number of input connections.
    n = fan_in
  elif mode='FAN_OUT': # Count only number of output connections.
    n = fan_out
  elif mode='FAN_AVG': # Average number of inputs and output connections.
    n = (fan_in + fan_out)/2.0

    truncated_normal(shape, 0.0, stddev=sqrt(factor / n))
```

This made me think that the default was `truncated_normal`. 
And in the code I found the same thing:
https://github.com/tensorflow/tensorflow/blob/v1.15.0/tensorflow/contrib/layers/python/layers/initializers.py#L149

Reply:
Nice! After looking at the code I think you are right, the value should be `truncated_normal` (the default). 
Even though I was facing diverging results, so I did an additional experiment:

IN TF 1.15:
```Python
ini = tf.contrib.layers.variance_scaling_initializer(seed=6)
with tf.compat.v1.Session() as sess:  print(ini([10]).eval())
```
`[ 0.3352677, 0.3420339, 0.18761884, 0.1507059, 0.56236863, -0.46897727, 0.15543926, -0.33059058, -0.71044177, 0.14339857]`

In TF 2.3.1

```Python
ini = tf.keras.initializers.VarianceScaling(scale=2.0, seed=6)
ini([10])
```
`[ 0.6524273, -0.2395821,  0.2508376, -0.11647213, -0.6710456 , 0.74666166,  0.9594222,  0.16396938,  0.4017174,  0.3574336 ]`

An then,
```Python
ini = tf.compat.v1.initializers.variance_scaling(scale=2.0, seed=6)
ini([10])
```
`[ 0.33428904,  0.3410355 ,  0.18707119,  0.15026599,  0.56072706, -0.46760836,  0.15498553, -0.3296256 , -0.708368,  0.14298]`

The last method returns almost the same values as in TF 1.15, so I think we should use it instead.
I've also check it in my experimental setup and the results are closer to the ones reported with TF 1.15

However, I couldn't find a satisfactory explanation. The implementations seem very similar ([v1](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/init_ops.py), [v2](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/init_ops_v2.py)).

Maybe the only difference is the random engine?

Reply:
Good work! I must have missed `tf.compat.v1.initializers.variance_scaling` as a possiblitliy! 
I adapted to use the `compat` function!